### PR TITLE
Action check broken links

### PR DESCRIPTION
**Pull request by kouliavtsev**
Adds a GitHub action that checks for broken links.

- Uses `zentered/vercel-preview-url@v1.0.0` to retrieve Vercel Preview URL.
- Passes the URL into link-check.mjs
- Checks for broken links with `markdown-link-check`

### CODE REVIEW

**Comment by kouliavtsev** [comment]
@noccor 

So, far implemented ` link-check.mjs` that only work locally. 

The `markdown-link-check` throws an error if the URL does not have a forward slash when it is an internal link.

`[Example](example.html)` 🛑 Good
`[Example](/example.html)` ✅ Bad

This has been fixed. 

After running the script, I found mucho dead links. Let's fix this before we implement this as GitHub action.

```
#how-do-i-filter-data-that-s-sent-to-appsignal is dead 💀
http://localhost:3000/api/index.html is dead 💀
http://localhost:3000/graphql-explorer-examples.html is dead 💀
#every-occurance is dead 💀
#nevery-notify is dead 💀
#retries-footnote is dead 💀
#language-libries is dead 💀
#working-directory-location is dead 💀
#working-directory-permissions is dead 💀
#alerting is dead 💀
#alerts is dead 💀
#performance-issue is dead 💀
#node.js is dead 💀
https://80beans.campfirenow.com/f08a4 is dead 💀
http://localhost:3000/platforms/integrations is dead 💀
http://localhost:3000/formatting.html#plaintext is dead 💀
http://localhost:3000/formatting.html#logfmt is dead 💀
http://localhost:3000/formatting.html#json is dead 💀
http://localhost:3000/formatting.html#ndjson is dead 💀
http://localhost:3000/formatting.html is dead 💀
#platform-configuraiton is dead 💀
http://localhost:3000/platforms/cloudwatch.html is dead 💀
http://localhost:3000/platforms/clevercloud.html is dead 💀
http://localhost:3000/platforms/gigalixir.html is dead 💀
http://localhost:3000/platforms/heroku.html is dead 💀
http://localhost:3000/platforms/http.html is dead 💀
http://localhost:3000/platforms/netlify.html is dead 💀
http://localhost:3000/platforms/syslog.html is dead 💀
http://localhost:3000/log-management is dead 💀
http://localhost:3000/troubleshooting is dead 💀
#Filtering-and-Querying is dead 💀
#Search-Query-Overview is dead 💀
#Edit-Source is dead 💀
#Environment-Variables is dead 💀
#Command-Line is dead 💀
#metric-values-file-size is dead 💀
http://localhost:3000/known-issues/gem-instrumentation-compatibility.html is dead 💀
http://localhost:3000/known-issues/puma-phased-restart.html is dead 💀
http://localhost:3000/known-issues/ruby-2-6-extension-not-installing.html is dead 💀
http://localhost:3000/known-issues/compilation-dependencies-required-at-runtime.html is dead 💀
http://localhost:3000/known-issues/shutdown-of-push-api-version-1.html is dead 💀
http://localhost:3000/known-issues/elixir-otp-21.html is dead 💀
http://localhost:3000/known-issues/rails-5-1-missing-errors.html is dead 💀
http://localhost:3000/known-issues/plug-actions-registered-as-unknown.html is dead 💀
http://localhost:3000/known-issues/incorrect-container-host-metrics.html is dead 💀
http://localhost:3000/known-issues/alpine-linux-ruby-gem-2-4-elixir-package-1-4-upgrade-problems.html is dead 💀
http://localhost:3000/known-issues/sidekiq-delayed-extension-action-names.html is dead 💀
http://localhost:3000/deploy-markers.html is dead 💀
http://localhost:3000/custom-markers.html is dead 💀
http://localhost:3000/demo.html is dead 💀
#default is dead 💀
#system is dead 💀
#file is dead 💀
#env is dead 💀
#override is dead 💀
http://localhost:3000/parameter-filtering.html is dead 💀
#appsignal-send_error-3 is dead 💀
http://localhost:3000/phoenix.html is dead 💀
http://localhost:3000/configuration.html#creating-a-log-source is dead 💀
#remove-integration-packages is dead 💀
#initial is dead 💀
http://localhost:3000/event-formatters.html is dead 💀
#adding-your-own-probe is dead 💀
http://localhost:3000/delayed-job.html is dead 💀
http://localhost:3000/que.html is dead 💀
http://localhost:3000/resque.html is dead 💀
http://localhost:3000/sidekiq.html is dead 💀
http://localhost:3000/active-job.html is dead 💀
#minutely-probe-configuration is dead 💀
#minutely-probe-metrics is dead 💀
#minutely-probe-configuration-secrets is dead 💀
#appsignal-stop-requirement is dead 💀
http://localhost:3000/dns-timeouts.html is dead 💀
#appsignal-send_error-6 is dead 💀
#helper-transactions is dead 💀
https://github.com/appsignal/appsignal-nodejs/blob/main/packages/nodejs/src/config.ts is dead 💀
http://localhost:3000/scopes.html is dead 💀
#request-session-data is dead 💀
```

To get the script working locally:
- Start website with `yarn dev` (or `yarn build` + `yarn start`)
- run `link:check`

**Comment by kouliavtsev** [comment]
Learned that `yarn link:check` does not work well with `yarn start`. When you run it it marks all `src` of the images invalid.

It does work with `yarn dev`.

Investigating... 🕵️‍♂️

**Comment by tombruijn** [comment]
@kouliavtsev I think I configured it correctly. Let me know if it doesn't work.

**Comment by kouliavtsev** [comment]
@tombruijn It works now! 🎉

Screenshot 2022-11-28 at 16 50 29

**Comment by kouliavtsev** [comment]
@noccor 

The https://github.com/appsignal/appsignal-docs/actions/runs/3573665182/jobs/6007988363 is fixed now. I did some digging and learned that footnotes are part of `Extended Markdown Syntax`, which we support. :)

Here is an example of how it works. In our case, we used HTML in MDX, which we IMO. could avoid.

```mdx
Here's a simple footnote,[^1] and here's a longer one.[^bignote]

[^1]: This is the first footnote.

[^bignote]: Here's one with multiple paragraphs and code.

    Indent paragraphs to include them in the footnote.

    `{ my code }`

    Add as many paragraphs as you like.
```

https://www.markdownguide.org/extended-syntax/#footnotes

Here you can see the footnote updated in the docs. https://github.com/appsignal/appsignal-docs/pull/845/commits/fed61cca890cb33573204845c10169e2be3b20f9

While implementing, I have learned that you can define a footnote only once. We are using it 3 times in the current doc.
 
Screenshot 2022-11-29 at 16 03 06

In other words, it is not possible to link to footnote `[^1]` a second or third time.

To solve this, can we link these as footnote `1`, `2`?

Another solution could be adding HTML like this, which is not ideal.

Screenshot 2022-11-29 at 16 06 33
